### ISSUE TITLE
Accessibility: Displaying related content above its heading

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyfive#150]
**Description**

This is difficult for me to describe and accessibility experts would be able to correct and add more context.

Heading elements are used to structure content, adding a hierarchy of importance, starting with H1.  https://www.w3.org/TR/WCAG20-TECHS/H69
They can also be used to navigate page content.

Besides headings, there are other ways to structure content, but I want to get to the point.
All the content also needs to be inside a landmark, in this theme that includes: `<header>`, `<footer>`, `<nav>`,`<main>`.
The theme does not currently use `<article>`, `<aside>` or `<section>`. 

A heading is used to describe the content that comes after it.
Content that is placed before a heading is not semantically associated with the heading that comes after.

Picture the following HTML structures
```
header 
 navigation

main
H1
  list
   list item
    paragraph
    H2 (post title)
   list item
    paragraph
    H2 (post title)
      H3 (heading inside post content)
  list
```

```
header 
 H1
 navigation

main
    paragraph
    H2 (post title)
    paragraph
    H3 (heading inside post content)
      H4 (heading inside post content)
```

The first paragraph is associated with the H1.
The second paragraph is associated with the first H2.

So depending on how you are navigating the page content, the category is associated with the wrong post,
and this can make it difficult for users to find the content they need.

```
Kittens (category)
Hello World (post title)

puppies (category)
Hello WordPress (post title)
```

By the rule of proximity and visually, I will speculate that many users will be able to tell that Hello World belongs in the category "Kittens".
But semantically, and for example for some users of screen readers, "kittens" isn't associated with any post. Hello World is associated with "puppies".

If a screen reader user reads all the content of the blog page, they will be alerted that these blogs are **separate list items**,
and it may be easier to understand the context that the paragraph and heading belongs together. 
But if a user navigates content by headings, they will not receive the information about which category the post belongs in, because they will not know that once they select the heading, they actually need to go **backwards** to find and read the category.

Similarly on single post and pages, a paragraph above the post title would either be "orphaned" or be associated with a heading that is above it, not the post title below it. This is less confusing than in a the blog scenario, since there is a lower risk of mix-ups.

------

Let's look at some options:

# Placing content in the correct position in the DOM but moving it with CSS.
This is not an option because especially if the moved item is focusable, like a category name that is a link,
the visual order and the DOM must match, or it will be difficult to navigate using the keyboard, a screen reader, magnifier, etc.

# Wrapping each post in the post list in article
To my understanding, on a blog, this would have the same affect as grouping the posts using the list, which the query loop / post list block already does. So it is not a solution.

# Using aria to associate the paragraph with the heading
A last resort could be using aria to make the linked post title be described by the category.
But the block editor does not support this, it would cause a block validation error.

# Hide content of little importance from screen readers
Hiding elements using custom CSS could be an option in a scenario where the text has a decorative purpose and does not add necessary context.
```
<p>Stories</p>
<h2>Tell your story</h2>
```
It becomes a problem as soon as the text has importance, such as adding a category or branding.

**Comment by afercia** [comment]
As @carolinan pointed out, placing important content before the heading would inevitably make that content completely missed for some users. Most notably, blind screen reader users who jump through headings won't have a clue there's something relevant _before_ the heading.

All WordPress bundled themes are meant to show off the best of WordPress and the new release features. They are meant to show off best practices and the quality of the content that can be published by using WordPress. That includes accessibility. They also have an educational purpose, as users and developers would learn from the best practices of a bundled theme.

As such, I'd expect any WordPress bundled theme to ship with the highest level of accessibility.

I'd kindly ask to just change the design and move any post-related information like the category (or post meta or any related post content) _after_ the heading.

Also, I'm not sure I understand why the category should come first and the tags after.

I think there have been very few exceptions to this pattern in other bundled themes, for example:
- Twenty Twenty: shows the category before the heading. As said, this is less than ideal.
- Twnenty Seventeen: shows the post meta before the heading. Also not ideal.

**Comment by jasmussen** [comment]
Judging from the source designs, and #195, the pill shape plays an important role. Keeping in mind the suggested need to move any post-related information below the heading, I did a quick audit of the designs in question.

**Testimonial:**

![testimonial](https://github.com/user-attachments/assets/aa91bda6-7cbc-40ba-b2f1-5d6a6b943b7a)

This is a pattern with a quote, and is part of the content.

&nbsp;

**Podcast landing:**

![podcast](https://github.com/user-attachments/assets/fa06d0c3-f375-40ec-919d-a7be08e3f1a9)

Pills are used as buttons, part of the content.

&nbsp;

**Tags:**

![tags](https://github.com/user-attachments/assets/9ae09695-df62-4288-86dc-b855cef3335d)

Tags are below the post.

&nbsp;

**Explore:**

![explore](https://github.com/user-attachments/assets/fa7a3577-6465-4e0d-8357-c0d9b393a2d5)

Taxonomy is to the right of the heading, semantically *after* the heading.

&nbsp;

**Membership:**

![membership](https://github.com/user-attachments/assets/b5a7c439-c7f7-4c47-bcf4-0afecf776e82)

Decorative chip used semantically after the heading, but it's also a pattern.

&nbsp;

**About us:**

![about us](https://github.com/user-attachments/assets/803e6fb8-b584-4dc6-8776-5c35f1128cc8)

This is also a pattern, meant to be a section of a larger page.

&nbsp;

**Become a member:**

![become a member](https://github.com/user-attachments/assets/a49b12f1-e11a-4b13-892c-029c2592b08d)

Same as above, as a pattern meant to be part of a larger page.

&nbsp;

**Tell your story:**

<img width="1440" alt="stories" src="https://github.com/user-attachments/assets/ac99019e-b4b6-44f7-874e-420023883053">

This appear to be the primary source of feedback, given the chip above.

&nbsp;

Moving the chip below does lose some character:

<img width="1440" alt="stories-alt" src="https://github.com/user-attachments/assets/47de06b8-0532-4ef9-b7af-5947da8cc0ec">

&nbsp;

Another option is removing that chip from this particular pattern. It seems solvable, though. Any preferences, @beafialho

**Comment by beafialho** [comment]
Thank you for everyone's inputs. Given that the "Tell your story" section, part of the Photo blog template, seems to be the main one issue to solve here, I think it'd be okay to move it below like @jasmussen's example:

![Captura de ecrã 2024-09-03, às 15 50 25](https://github.com/user-attachments/assets/45782b00-3709-4458-9fd4-a574806c43e7)

Hopefully this solves it.

**Comment by carolinan** [comment]
> Same as above, as a pattern meant to be part of a larger page.

That there may be headings above it, inside content or another pattern, does not solve the issue, rather it makes it more confusing and more difficult to understand the structure of the page.
The content will be associated with the wrong heading:

H1 Hello World
H2 The best apple pie recipe
Become a member
H2 Subscribe to...

Compared to:
H1 Hello World
H2 The best apple pie recipe
H2 Become a member: Subscribe to...


H1 Hello World
H2 The best apple pie recipe
H2 Subscribe to...
Become a member

**Comment by afercia** [comment]
I agree with @carolinan that the order must be consistent to be understandable. Buttons, chips, or any other 'meta information' that is _sometimes_ placed after the heading and _sometimes_ placed before it wouldn't help understand the structure of the content. I think it's important to take into account other surrounding content and not see the patterns 'in isolation'.

In the screenshot below, a very rough visual example of what @carolinan pointed out. To what content 'Become a member' is associated to?

![become a member](https://github.com/user-attachments/assets/d0928713-7b46-4c77-92b1-ce9309801210)

To avoid any potential confusion, there should be a consistent pattern everywhere.

**Comment by richtabor** [comment]
> In the screenshot below, a very rough visual example of what @carolinan pointed out. To what content 'Become a member' is associated to?

There should be more spacing between the two sections. I wouldn't think this would be an acceptable orientation of content.

**Comment by carolinan** [comment]
@richtabor I think you missed the point. A user without vision will not experience the space.

**Comment by afercia** [comment]
I'd support @carolinan argument that _blind_ screen reader users would have some troubles to take advantage from some more visible white space.
